go-controller/pkg/cluster/master.go

```diff
@@ -260,15 +236,28 @@ func (cluster *OvnClusterController) ensureNodeLogicalNetwork(nodeName string, h
 }
 
 func (cluster *OvnClusterController) addNode(node *kapi.Node) (err error) {
-	var hostsubnet *net.IPNet
+	hostsubnet, _ := parseNodeHostSubnet(node)
+	if hostsubnet != nil {
+		// Node already has subnet assigned; ensure its logical network is set up
+		return cluster.ensureNodeLogicalNetwork(node.Name, hostsubnet)
```

Comment:
The new version of the code is definitely clearer, but my previous question still stands: if the hostsubnet annotation is set, shouldn't that imply that the logical network was already set up by the previous call to `addNode` which created the annotation?

Reply:
@danwinship not necessarily, if for example the ovnkube pod completely dies, that wipes the OVN databases (since we want to reconstruct entirely on startup), then everything restarts. In that case we'll have an annotation but no logical network.